runtime/databricks/automl_runtime/sklearn/base_datetime_transformer.py

```diff
@@ -36,16 +40,69 @@ class BaseDatetimeTransformer(ABC, TransformerMixin, BaseEstimator):
     DAYS_IN_YEAR = 366  # Account for leap years
     WEEKEND_START = 5
 
+    def __init__(self, impute_method=None):
+        """Create a `BaseDatetimeTransformer`.
+
+        Parameters
+        ----------
+        impute_method: method used for imputation, with the following options:
+            'mean', 'median', 'most_frequent', or a string that can be converted to
+            a date or timestamp by `pd.to_datetime` function.
+            If not specified, the default is "1970-01-01" (unix epoch).
+        """
+        if not impute_method:
+            impute_method = self.EPOCH
```

Comment:
EPOCH is not a method. What is the purpose to set the value to EPOCH?

Reply:
Should we consider to change the default method?

Reply:
I agree here the "method" is not the best naming, as users can pass in either "mean" or a constant. `EPOCH` is a constant `1970-01-01`, which was the default value.

For default method, I kept `1970-01-01` to be backward compatible, and feel it's not a very bad option. It essentially signifies the date is missing, and our featurization will tell it apart from regular dates. With that said, I think "median" or "most_frequent" are equally good. Let me know what you feel is the best default.

Reply:
So if the impute_method is a string, it means is it a constant imputation? Then I would suggest to change the name to make it more clear. Or add a const imputation method with extra input.